src/js/common/components/Search/SearchBar2024.jsx

```diff
@@ -73,6 +73,11 @@ class SearchBar2024 extends Component {
     const dataLayerObject = {
       event: 'searchKeyword',
       searchKeyword: searchString,
+      source:{
+        pageName: window.location.href,
+        pageType: "challenge",
```

Comment:
We want the pageName & pageType to come from the parent component that calls `searchBar20204`, but instead of adding props we need to pass from the parent, I think another approach will be more maintainable. Please create a function in `/js/utils/calculateGTMPageType.js` which takes in the pathName, and returns one of our pageTypes: 
ballot, politician, candidate, donation, homepage, ballot, account, challenge, issue
...and then an even more specific pageName. Feel free to reach out with questions!